pom.xml

```diff
@@ -798,13 +798,21 @@ under the License.</licenseText>
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-pmd-plugin</artifactId>
-          <dependencies>
-            <dependency>
-              <groupId>net.sourceforge.pmd</groupId>
-              <artifactId>pmd-core</artifactId>
-              <version>7.14.0</version>
-            </dependency>
-          </dependencies>
+          <configuration>
+            <analysisCache>true</analysisCache>
+            <analysisCacheLocation>${maven.multiModuleProjectDirectory}/.pmd/.cache</analysisCacheLocation>
```

Comment:
```suggestion
            <analysisCache>true</analysisCache>
            <analysisCacheLocation>${maven.multiModuleProjectDirectory}/.pmd/.cache</analysisCacheLocation>
```
would give here `CoC` value as well, to avoid confusion.
is central cache good or better dedicated to each module.

Having both variants implied, so one way is wrong already, as done different, assuming this valid as not mentioned by you @adangel.

Giving nifty **default value** to `analysisCacheLocation` wont hurt, right?

Reply:
The defaults are documented in the documentation (surprise!) https://maven.apache.org/plugins/maven-pmd-plugin/pmd-mojo.html#analysisCache

Not sure, what "maven.multiModuleProjectDirectory" would resolve to - probably to the same directory for every single module. Hence this config probably results in overwriting the cache file on every run - and the cache is useless.

Did you try whether the cache is working? Running "./mvnw clean verify" and again "./mvnw verify" (without clean) and comparing the timings of the pmd plugin?

The default value seems to be more sensible, as it uses a cache file per module.

Reply:
ty.

Reply:
2 sec.

<img width="744" alt="image" src="https://github.com/user-attachments/assets/8c194adb-5c2a-4eb3-b17f-7df97c2639c4" />



clean. 


init takes 4 sec.

<img width="673" alt="image" src="https://github.com/user-attachments/assets/e0bed4b3-194f-44b5-bb1c-dcebace8e801" />



switch working 

50% gain:

<img width="718" alt="image" src="https://github.com/user-attachments/assets/b55ffad0-cde9-46a5-8aba-7afcc68fdb58" />









<img width="769" alt="image" src="https://github.com/user-attachments/assets/cc814d38-4b97-49a8-a16f-a372724189da" />

<img width="727" alt="image" src="https://github.com/user-attachments/assets/32527f3d-8297-4843-a4de-5b2dc6169b18" />